synapse/federation/federation_client.py

```diff
@@ -250,6 +250,8 @@ async def backfill(
 
         logger.debug("backfill transaction_data=%r", transaction_data)
 
+        assert transaction_data is not None
```

Comment:
The `backfill` method of the transport client suggests maybe we should never be returning `None` instead of this. 🤷 It looks like it isn't possible for it to return `None` though since we check `extremities` above. If nothing else, please add a comment about why this assertion is safe.